src/averageAges.js

```diff
@@ -55,7 +69,19 @@ function calculateWomenAverageAge(people, withChildren) {
  * @return {number}
  */
 function calculateAverageAgeDiff(people, onlyWithSon) {
-  // write code here
+  const children = onlyWithSon
+    ? people.filter(p => p.sex === 'm'
+      && people.some(m => m.name === p.mother))
+    : people.filter(p => people.some(m => m.name
+      === p.mother));
+
+  let sum = 0;
+
+  children.forEach((ch) => sum += ch.born - people.find(
+    (m) => m.name === ch.mother
+  ).born);
```

Comment:
In this snippet we can change its solution to use a `reduce` function, as it is a mor readable and more performant way to do it.

Storing its result in a variable is also a good practice.

```js
const averageDiff = children.reduce((acc, child) => { 
  const mother = people.find((m) => m.name === child.mother);
  return mother ? acc + (child.born - mother.born) : acc;
}, 0);
```